storage/writer.go

```diff
@@ -88,6 +88,15 @@ type Writer struct {
 	// cancellation.
 	ChunkRetryDeadline time.Duration
 
+	// ChunkTransferTimeOut sets a per-chunk retry transfer timeout for multi-chunk
+	// resumable uploads.
+	//
+	// For uploads of larger files, the Writer will attempt to retry if the
+	// request to upload a particular chunk stalls for some time.
+	// If a single chunk has been attempting to upload for longer than this
+	// timeout, then it will be retried. The default value no timeout.
+	ChunkTransferTimeOut time.Duration
```

Comment:
Let's keep capitalization consistent with the other repo; use `ChunkTransferTimeout`